scripts/zones/King_Ranperres_Tomb/npcs/Tombstone.lua

```diff
@@ -22,18 +22,17 @@ end
 
 function onTrigger(player, npc)
     local currentMission = player:getCurrentMission(SANDORIA)
-    local MissionStatus = player:getCharVar("MissionStatus")
-    local BatHuntCompleted = player:hasCompletedMission(SANDORIA, tpz.mission.id.sandoria.BAT_HUNT) -- quest repeatable and clicking tombstone should not produce cutscene on repeat
+    local missionStatus = player:getCharVar("MissionStatus")
     local X = npc:getXPos()
     local Z = npc:getZPos()
 
     if X >= -1 and X <= 1 and Z >= -106 and Z <= -102 then
-        if currentMission == tpz.mission.id.sandoria.BAT_HUNT and MissionStatus <= 1 then
+        if currentMission == tpz.mission.id.sandoria.BAT_HUNT and missionStatus <= 1 then
             player:startEvent(4)
         else
             player:startEvent(2)
         end
-    elseif currentMission == tpz.mission.id.sandoria.RANPERRE_S_FINAL_REST and MissionStatus == 2 then
+    elseif currentMission == tpz.mission.id.sandoria.RANPERRE_S_FINAL_REST and missionStatus == 3 then
         player:startEvent(8)
     end
```

Comment:
Is there anything preventing the player from repeating this CS every time they click this?

Reply:
no, thanks for pointing that out... it needs to check if they have the KI or not, since that's what flags the next part